libp2p/muxers/muxer.nim

```diff
@@ -67,9 +67,9 @@ method init(c: MuxerProvider) =
       if not isNil(c.muxerHandler):
         futs &= c.muxerHandler(muxer)
 
-      checkFutures(await allFinished(futs))
-    except CancelledError as exc:
-      raise exc
+      await allFuturesThrowing(futs)
```

Comment:
hm, I wonder why would this leak here at all, we still don't do anything if any of them raised, because we need to handle all errors in the `handler` proc.